File: pkg/model/stock_out.go
package model

import "time"

type StockOut struct {
	ID          uint
	CreatedBy   User
	Items       []StockOutItem
	Status      string
	CreatedAt   time.Time
	UpdatedAt   time.Time
	FinalizedAt *time.Time
}

type StockOutItem struct {
	ID            uint
	StockOutID    uint
	Item          Item
	TotalQuantity float64
	Packagings    []StockOutPackaging
}

type StockOutPackaging struct {
	ID             uint
	StockOutItemID uint
	ItemPackaging  ItemPackaging
	Quantity       int
}

File: client/grao-a-grao/src/util/factory/stock_out.ts
import {
    StockOutModel,
    StockOutItemModel,
    StockOutPackagingModel,
  } from "@/types/stock_out";
  
  /**
   * Creates a new, empty packaging entry for a StockOutItem.
   */
  export function createEmptyStockOutPackaging(): StockOutPackagingModel {
    return {
      id: undefined,
      item_packaging: {
        id: undefined,
        description: "",
        quantity: 0,
      },
      quantity: 1, // default to one packaging unit
    };
  }
  
  /**
   * Creates a new, empty StockOutItem with a single empty packaging.
   */
  export function createEmptyStockOutItem(): StockOutItemModel {
    return {
      id: undefined,
      item: {
        id: undefined,
        description: "",
        is_fractionable: true,
      },
      total_quantity: 0,
      packagings: [createEmptyStockOutPackaging()],
    };
  }
  
  /**
   * Creates a new, empty StockOut with one empty item.
   */
  export function createEmptyStockOut(): StockOutModel {
    return {
      id: undefined,
      status: "draft",
      items: [createEmptyStockOutItem()],
      created_at: undefined,
      updated_at: undefined,
    };
  }
  